compiler/rustc_mir_build/src/build/mod.rs

```diff
@@ -403,6 +403,15 @@ enum NeedsTemporary {
 #[must_use = "if you don't use one of these results, you're leaving a dangling edge"]
 struct BlockAnd<T>(BasicBlock, T);
 
+impl BlockAnd<()> {
+    /// Unpacks `BlockAnd<()>` into a [`BasicBlock`].
+    #[must_use]
+    fn unpack_block_and_unit(self) -> BasicBlock {
```

Comment:
This just bikeshedding but we could orientate ourselves by [`Result::into_ok`](https://doc.rust-lang.org/nightly/std/result/enum.Result.html#method.into_ok) and go with `into_block`. `Result::into_ok` is similar to this method in the sense that it "discharges" a "meaningless piece of information" from a more complex thing (`into_ok`: sum type w/ `!`, `into_block`: product type w/ `(=`). And doesn't mention `never` in its name.

Reply:
Yeah, this sounds like an improvement to me. I'll try it out.